src/hazardous/mac/poly1305.rs

```diff
@@ -143,93 +148,29 @@ impl core::fmt::Debug for Poly1305 {
+        // One byte is appended to detect trailing zeroes if not last chunk.
+        mb[16] = if self.is_finalized { 0 } else { 1 };
+        let mut m: fiat_poly1305_tight_field_element = [0u32; 5];
+        fiat_poly1305_from_bytes(&mut m, &mb);
 
-        let r0: u32 = self.r[0];
-        let r1: u32 = self.r[1];
-        let r2: u32 = self.r[2];
-        let r3: u32 = self.r[3];
-        let r4: u32 = self.r[4];
-
-        let s1: u32 = r1 * 5;
-        let s2: u32 = r2 * 5;
-        let s3: u32 = r3 * 5;
-        let s4: u32 = r4 * 5;
-
-        let mut h0: u32 = self.a[0];
-        let mut h1: u32 = self.a[1];
-        let mut h2: u32 = self.a[2];
-        let mut h3: u32 = self.a[3];
-        let mut h4: u32 = self.a[4];
-
-        // h += m[i]
-        h0 += (load_u32_le(&data[0..4])) & 0x3ffffff;
-        h1 += (load_u32_le(&data[3..7]) >> 2) & 0x3ffffff;
-        h2 += (load_u32_le(&data[6..10]) >> 4) & 0x3ffffff;
-        h3 += (load_u32_le(&data[9..13]) >> 6) & 0x3ffffff;
-        h4 += (load_u32_le(&data[12..16]) >> 8) | hibit;
-
-        // h *= r
-        let d0: u64 =
-            (h0 as u64 * r0 as u64) +
-            (h1 as u64 * s4 as u64) +
-            (h2 as u64 * s3 as u64) +
-            (h3 as u64 * s2 as u64) +
-            (h4 as u64 * s1 as u64);
-        let mut d1: u64 =
-            (h0 as u64 * r1 as u64) +
-            (h1 as u64 * r0 as u64) +
-            (h2 as u64 * s4 as u64) +
-            (h3 as u64 * s3 as u64) +
-            (h4 as u64 * s2 as u64);
-        let mut d2: u64 =
-            (h0 as u64 * r2 as u64) +
-            (h1 as u64 * r1 as u64) +
-            (h2 as u64 * r0 as u64) +
-            (h3 as u64 * s4 as u64) +
-            (h4 as u64 * s3 as u64);
-        let mut d3: u64 =
-            (h0 as u64 * r3 as u64) +
-            (h1 as u64 * r2 as u64) +
-            (h2 as u64 * r1 as u64) +
-            (h3 as u64 * r0 as u64) +
-            (h4 as u64 * s4 as u64);
-        let mut d4: u64 =
-            (h0 as u64 * r4 as u64) +
-            (h1 as u64 * r3 as u64) +
-            (h2 as u64 * r2 as u64) +
-            (h3 as u64 * r1 as u64) +
-            (h4 as u64 * r0 as u64);
-
-        // (partial) h %= p
-        let mut c: u32 = (d0 >> 26) as u32; h0 = (d0 & 0x3ffffff) as u32;
-        d1 += c as u64; c = (d1 >> 26) as u32; h1 = (d1 & 0x3ffffff) as u32;
-        d2 += c as u64; c = (d2 >> 26) as u32; h2 = (d2 & 0x3ffffff) as u32;
-        d3 += c as u64; c = (d3 >> 26) as u32; h3 = (d3 & 0x3ffffff) as u32;
-        d4 += c as u64; c = (d4 >> 26) as u32; h4 = (d4 & 0x3ffffff) as u32;
-        h0 += c * 5; c = h0 >> 26; h0 = h0 & 0x3ffffff;
-        h1 += c;
-
-        self.a[0] = h0;
-        self.a[1] = h1;
-        self.a[2] = h2;
-        self.a[3] = h3;
-        self.a[4] = h4;
+        // h += m
+        let mut h: fiat_poly1305_loose_field_element = [0u32; 5];
+        fiat_poly1305_add(&mut h, &self.a, &m);
+        // h *= r with partial reduction modulo p
```

Comment:
```rust
// h += m
```

```rust
// h *= r with partial reduction modulo p
```

These were very helpful comments for my understanding. 👍🏼

Reply:
Their actually leftovers from the Andrew Moon adaption. But yeah, I thought we'd just keep them there.